common/djangoapps/entitlements/models.py

```diff
@@ -219,3 +219,7 @@ def set_enrollment(cls, entitlement, enrollment):
         Fulfills an entitlement by specifying a session.
         """
         cls.objects.filter(id=entitlement.id).update(enrollment_course_run=enrollment)
+
+    @classmethod
+    def entitlements_for_user(cls, user):
+        return cls.objects.filter(user=user, expired_at=None).select_related('user')
```

Comment:
This might need to be updated.  What is a user is enrolled, but the entitlement has expired? Does the method name need to be changed to reflect that this will only return only unexpired entitlements regardless of enrollment status.

Reply:
the CourseEnrollment equivalent classmethod  enrollments_for_user only retrieves active enrollments for the user, so I figured unexpired entitlements would be comparable. We are retrieving only those entitlements for the user that are currently relevant

Reply:
These two are not equivalent.  A user can have a Course Entitlement that is still active even with the expiration set.  An Entitlement is active is an enrollment is set and not NULL regardless of expiration date.